lib/tools/system-calls.js

```diff
@@ -727,20 +721,27 @@ systemCallMethods.checkAvdExist = async function (avdName) {
  * @throws {Error} If the emulator is not ready within the given timeout.
  */
 systemCallMethods.waitForEmulatorReady = async function (timeoutMs = 20000) {
-  let start = Date.now();
-  log.debug("Waiting until emulator is ready");
-  while ((Date.now() - start) < timeoutMs) {
-    try {
-      let stdout = await this.shell(["getprop", "init.svc.bootanim"]);
-      if (stdout.indexOf('stopped') > -1) {
-        return;
+  try {
+    await waitForCondition(async () => {
+      try {
+        if (!await this.shell(['getprop', 'init.svc.bootanim']).includes('stopped')) {
+          return false;
+        }
+        // Sometimes the package manager service might still being initialized
+        // on slow systems even after emulator booting is completed
+        const installLocation = await this.shell(['pm', 'get-install-location']);
+        return /\d+\[\w+\]/.test(installLocation);
```

Comment:
Some explanation of what this regular expression is checking would be good.

Reply:
ok will add